devstack/Makefile

```diff
@@ -0,0 +1,34 @@
+DEVSTACK_COMPOSE_FILES=-f devstack/devstack-compose-ocim.yml.tmp
+
+ifeq ($(firstword $(MAKECMDGOALS)),$(filter $(firstword $(MAKECMDGOALS)),devstack.shell))
+  RUN_ARGS := $(wordlist 2,$(words $(MAKECMDGOALS)),$(MAKECMDGOALS))
+  $(eval $(RUN_ARGS):;@:)
+endif
+
+devstack.up: devstack.build devstack/honcho.env.test.tmp devstack/honcho.env.tmp
+	docker-compose ${DEVSTACK_COMPOSE_FILES} up -d
+
+devstack.shell: devstack.up ## Run a command in devstack (or an interactive shell)
+	docker-compose ${DEVSTACK_COMPOSE_FILES} exec --user ${USER} \
+						 ocim-${USER} /usr/local/bin/shell.sh ${RUN_ARGS}
+
+devstack.logs: ## Show the logs of the devstack
+	docker-compose ${DEVSTACK_COMPOSE_FILES} ${subst devstack.,,$@}
+
+devstack.down: ## Destroy the devstack and all it contains
+	docker-compose ${DEVSTACK_COMPOSE_FILES} down
+	rm -f devstack/*.tmp
+
+DEVSTACK_BUILD=docker-compose ${DEVSTACK_COMPOSE_FILES} build
+
+devstack.build: devstack/devstack-compose-ocim.yml.tmp  ## Rebuild the devstack docker image
```

Comment:
A provisioning command can replace this build method.

Reply:
there will be a need for a `docker build` even if it is followed by an ansible run because the xenial image does not come with a ssh server or an unprivileged user that has sudo rights.